### PR TITLE
create data layer for worldpop_age_sex

### PR DESCRIPTION
**Pull request by tedw0ng**
Create data layer for Worldpop disaggregated data by age and sex groups.
Question: the regular worldpop layer has a .mean() line. Should it be sum()? Same question would apply to this new layer.

### CODE REVIEW

**Comment by S-AI-F** [comment]
@tedw0ng Agree that it should be sum() instead of mean(). I checked the original notebook and we are using also mean(): https://github.com/wri/cities-indicators/blob/emackres-patch-1/notebooks/extract-layers/extract-WorldPop-OpenSpaceTreeAccess.ipynb
I don't remember any reason for that. I think we should change this!

**Comment by S-AI-F** [comment]
I also see that worldpop provides the data through a specific api now: https://www.worldpop.org/sdi/introapi/
Should we consider changing the data access?

**Comment by tedw0ng** [comment]
I think the Worldpop API only returns JSON data, not rasters. That might be okay for indicators, but not for data layers.

**Comment by S-AI-F** [comment]
It actually serves also the url to get the tiff files but at the national level. So it is probably better if we keep extracting the data from GEE to get only areas of interest 
image

File: city_metrix/layers/worldpop_age_sex_class.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from dask.diagnostics import ProgressBar
+import xarray as xr
+import xee
+import ee
+
+from .layer import Layer, get_utm_zone_epsg, get_image_collection
+
+
+class WorldPop(Layer):
+    def __init__(self, **kwargs):
```
Comment (weiqi-tori) [inline city_metrix/layers/worldpop_age_sex_class.py:10]: `agesex_classes` should be added as an input parameter for `__init__()`

File: city_metrix/layers/worldpop_age_sex_class.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from dask.diagnostics import ProgressBar
+import xarray as xr
+import xee
+import ee
+
+from .layer import Layer, get_utm_zone_epsg, get_image_collection
+
+
+class WorldPop(Layer):
```
Comment (weiqi-tori) [inline city_metrix/layers/worldpop_age_sex_class.py:9]: Should give it a new class name

File: city_metrix/layers/worldpop_age_sex_class.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+		self.agesex_classes = agesex_classes
+
+    def get_data(self, bbox):
+        # load population
+        dataset = ee.ImageCollection('WorldPop/GP/100m/pop_age_sex')
+        world_pop = ee.ImageCollection(dataset
+                     .filterBounds(ee.Geometry.BBox(*bbox))
+                     .filter(ee.Filter.inList('year', [2020]))
```
Comment (weiqi-tori) [inline city_metrix/layers/worldpop_age_sex_class.py:19]: `year` would be more flexible if added as an input parameter with default value set to 2020

**Review by weiqi-tori — CHANGES_REQUESTED** [review]
Could add a test for the layer to check if it works well

File: city_metrix/layers/worldpop_age_sex_class.py (hunk 3 above)
Comment (tedw0ng) [reply city_metrix/layers/worldpop_age_sex_class.py:19]: added year as input parameter with default 2020

File: city_metrix/layers/worldpop_age_sex_class.py (hunk 2 above)
Comment (tedw0ng) [reply city_metrix/layers/worldpop_age_sex_class.py:9]: new class name is WorldPopAgeSex

**Comment by weiqi-tori** [comment]
The layer and test standard structure changed since this layer was created, so I fixed the conflict and updated the layer.

@tedw0ng: Please pull the branch with the latest update, and test if the layer functions as you expected. One note is that I switched the aggregation for image collection back to `mean()`. It's not aggregate the pixels within the bbox, it's for the image collection aggregating to an image when there is overlap between images.

**Comment by S-AI-F** [comment]
@weiqi-tori @chrowe 

Checked in this notebook and works globally well: https://colab.research.google.com/drive/1VtDgtQ_BVRh7h0QJ56ryfHJCByLs5Rqu#scrollTo=uXq9pqNOH7fw

Proposed changes:
- Would be great if we add an error checking on the "year" parameter. It should be 2020 and if the user specify different year, show an error message: "the selected year is not available". Now it works with any year value
- Can we add in the code the available age_sex list  of values: `[M_0, M_1, M_5, M_10, M_15, M_20, M_25, M_30, M_35, M_40, M_45, M_50, M_55, M_60, M_65, M_70, M_75, M_80, F_0, F_1, F_5, F_10, F_15, F_20, F_25, F_30, F_35, F_40, F_45, F_50, F_55, F_60, F_65, F_70, F_75, F_80]`

**Review by S-AI-F — APPROVED** [review]

**Review by weiqi-tori — APPROVED** [review]

**Pull request merged**